xcode.mk

```diff
@@ -7,10 +7,22 @@ CMAKE_BOOST_OPTIONS = -DBoost_NO_BOOST_CMAKE=TRUE \
 	-DBOOST_ROOT="$(BOOST_ROOT)"
 endif
 
+export OSX_SDK_PATH = $(shell xcrun --sdk macosx --show-sdk-path)
+
+# https://cmake.org/cmake/help/latest/envvar/MACOSX_DEPLOYMENT_TARGET.html
+# Make sure librime has the same deployment target with Squirrel
+# This prevent warnings like: `xxx was built for newer macOS version (xx) than being linked (xx)`
+export MACOSX_DEPLOYMENT_TARGET = 10.9
+
+CMAKE_OSX_OPTIONS = -DCMAKE_OSX_DEPLOYMENT_TARGET="$(MACOSX_DEPLOYMENT_TARGET)"
+
 ifdef BUILD_UNIVERSAL
-export CMAKE_OSX_ARCHS = -DCMAKE_OSX_ARCHITECTURES="arm64;x86_64"
```

Comment:
inspired by your last edit, setting environment variable `CMAKE_OSX_ARCHITECTURES` should also work.
but note this also require a recent version of cmake.